bazel/envoy_build_system.bzl

```diff
@@ -691,16 +691,16 @@ def envoy_select_google_grpc(xs, repository = ""):
 
 # Dependencies on Google grpc should be wrapped with this function.
 def envoy_google_grpc_external_deps():
-    return envoy_select_google_grpc([envoy_external_dep_path("grpc")])
+    return envoy_select_google_grpc([_envoy_external_dep_path("grpc")])
 
 # Select the given values if exporting is enabled in the current build.
-def envoy_select_exported_symbols(xs):
+def _envoy_select_exported_symbols(xs):
     return select({
         "@envoy//bazel:enable_exported_symbols": xs,
         "//conditions:default": [],
     })
 
-def envoy_select_force_libcpp(if_libcpp, default = None):
+def _envoy_select_force_libcpp(if_libcpp, default = None):
```

Comment:
I think most of the selects here are intended for export. If not, maybe we can regroup them elsewhere and give them a name with "internal" in it to avoid confusion with the exported select macros.

Reply:
The `envoy_select_foo` macros that I have marked as private with an `_` are all used only within `envoy_build_system.bzl`, at least within the Envoy code base (and google3 as well). Are you saying that we intentionally want to leave these as part of the public API of `envoy_build_system.bzl` so that external developers building upon Envoy may rely on them?

Reply:
No, I'm suggesting you move them to elsewhere in this file and rename them, since they are not really in the same category as the ones we are exporting, and it's confusing to colocate these in the same section of the file.

Reply:
Ah gotcha. I moved all the envoy_select_* macros to the bottom and append _internal to them. I'm planning a larger refactor of this file so I didn't move/rename all the private macros.